### PR TITLE
V7 1539 - Multi Labels Dataset

### PR DESCRIPTION
**Pull request by FrancescoSaverioZuppichini**
Since the APIs are not test-friendly, the following code is not tested. It should work, please have a look. `ClassificationDataset` now returns a Tensor of size 1 for single-label classification and a Tensor of size `[C,1]` for multi-label classification where `C` is the number of classes

### CODE REVIEW

File: darwin/torch/dataset.py [hunk 1]
```diff
@@ -108,17 +118,31 @@ def get_target(self, index: int):
...
+            # one hot encode all the targets
+            for tag in tags:
+                idx = self.classes.index(tag)
+                target[idx] = 1
+        else:
+            target = self.classes.index(tags[0])
+
         return target
 
+    def check_if_multi_label(self):
```
Comment (Fl4m3Ph03n1x) [inline darwin/torch/dataset.py:133]: def check_if_multi_label(self) -> None:

File: darwin/torch/dataset.py [hunk 2]
```diff
@@ -195,7 +219,9 @@ def get_target(self, index: int):
             # Extract the sequences of coordinates from the polygon annotation
             annotation_type = "polygon" if "polygon" in annotation else "complex_polygon"
```
Comment (Fl4m3Ph03n1x) [inline darwin/torch/dataset.py:220]: `annotation_type: str`

**Review by Fl4m3Ph03n1x — COMMENTED** [review]
Some documentation could be added. Other than that, LGTM

File: darwin/torch/dataset.py (hunk 1 above)
Comment (FrancescoSaverioZuppichini) [reply darwin/torch/dataset.py]: thanks, done!

File: darwin/torch/dataset.py (hunk 2 above)
Comment (FrancescoSaverioZuppichini) [reply darwin/torch/dataset.py]: thanks, done!

**Pull request merged**